lib/puppet/pops/types/type_asserter.rb

```diff
@@ -0,0 +1,42 @@
+module Puppet::Pops::Types
+  # Utility class for type assertion
+  #
+  class TypeAsserter
```

Comment:
Made comment on the commit - did not show up in the conversation. Repeating it here:
It is better to make this a Ruby module since there is no intention to ever instantiate this helper class, pass around as instances etc.